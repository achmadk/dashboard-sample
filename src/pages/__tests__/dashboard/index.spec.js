import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import chai from 'chai'

import DashboardPageAsyncContainer, {
  dashboardComponent
} from '@/pages/dashboard'
import DashboardPageContainer from '@/pages/dashboard/dashboard'
import ProviderStore from '@/../test/provider-store'

describe('Dashboard Page Async Container', () => {
  it('render correctly', () => {
    const wrapper = shallow(
      <ProviderStore>
        <DashboardPageAsyncContainer />
      </ProviderStore>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('return Dashboard component successfully', () => {
    const result = dashboardComponent()
    chai.expect(result).to.exist
  })
})

describe('Dashboard Page Component', () => {
  it('render correctly', () => {
    const wrapper = shallow(<DashboardPageContainer />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
