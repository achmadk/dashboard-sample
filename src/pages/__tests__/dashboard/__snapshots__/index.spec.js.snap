// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Page Async Container render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ConnectFunction />
</Provider>
`;

exports[`Dashboard Page Component render correctly 1`] = `
<div
  className="main-app-container"
>
  <SidenavContainer />
  <section
    className="app-page-container"
    id="page-container"
  >
    <HeaderContainer />
    <div
      className="app-content-wrapper"
    >
      <div
        className="app-content"
      >
        <div
          className="h-100"
        >
          <DashboardTypesComponent />
        </div>
      </div>
      <FooterContainer />
    </div>
  </section>
</div>
`;
